applications/launchpad/gui-react/src/components/Iterator/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { useTheme } from 'styled-components'
+
+import IconButton from '../IconButton'
+import Text from '../Text'
+import ArrowLeft from '../../styles/Icons/ArrowLeft2'
+import ArrowRight from '../../styles/Icons/ArrowRight2'
+
+import { Wrapper } from './styles'
+
+/**
+ * @name Iterator
+ * @description controlled presentation component for iterating over any value with next/previous buttons for the user
+ *
+ * @prop {string} value - current value
+ * @prop {() => void} next - callback for going to next value
+ * @prop {() => void} previous - callback for going to previous value
+ */
+const Iterator = ({
+  value,
+  next,
+  previous,
+  hasNext,
+  hasPrevious,
+}: {
+  value: string
+  next: () => void
+  previous: () => void
+  hasNext?: boolean
+  hasPrevious?: boolean
+}) => {
+  const theme = useTheme()
+
+  const disableNextButton = hasNext !== undefined && !hasNext
+  const disablePreviousButton = hasPrevious !== undefined && !hasPrevious
+
+  return (
+    <Wrapper>
```

Comment:
This should maybe be a fixed width wrapper so that the component doesn't shift so much as you cycle through the timeline:

https://user-images.githubusercontent.com/69508715/171397460-c9de0c17-4049-4262-84da-a2cf9493d964.mov

Reply:
On the Figma, I found that only 3 letters of the month are used, ie. April -> Apr:

![image](https://user-images.githubusercontent.com/11715931/171403090-f266c3d9-caa1-4ead-b0ef-b267aa65f327.png)

Reply:
changed to short month format

![Peek 2022-06-01 14-41](https://user-images.githubusercontent.com/9142942/171406870-e792f983-7dd6-4761-ba42-b087d1c7047d.gif)

Reply:
For three letters month, would it make sense to set fix width of this components? Still not the sleekest experience haviing this component flicker in size on click

Reply:
as above - when 'yearly' filter is selected, the labels change to just years "2022" and "2023", for example, but I guess we can fix different widths for both?

Reply:
changed to forced width:
![Peek 2022-06-01 21-42](https://user-images.githubusercontent.com/9142942/171488399-1a31b245-902c-4663-9910-
b534e4ece27a.gif)